core/src/main/java/org/apache/iceberg/BaseTransaction.java

```diff
@@ -728,4 +744,20 @@ TableOperations ops() {
   Set<String> deletedFiles() {
     return deletedFiles;
   }
+
+  /**
+   * Exception used to avoid retrying {@link PendingUpdate} when it is failed with {@link CommitFailedException}.
+   */
+  private static class ValidationFailureException extends RuntimeException {
```

Comment:
`ValidationFailureException` in `CatalogHandlers` is more about Illegal arguments. So, it is validation failure exception.

Here it is more of failed to apply pending update. Even though the purpose is to abort, they can have different explanations based on the context.

Hence renamed `ValidationFailureException` to `PendingUpdateFailedException` in BaseTransaction instead of making common code.